sycl/doc/extensions/SPIRV/SPV_INTEL_long_constant_composite.asciidoc

```diff
@@ -0,0 +1,226 @@
+_Constituents_ greater than maximum possible WordCount.
+
+The previous instruction must be an *OpSpecConstantComposite* or an
+*OpSpecConstantCompositeContinuedINTEL* instruction.
+
+_Constituents_ follow the same rules as defined for _Constituents_ of
+*OpSpecConstantComposite* instruction and specify members of a structure, or
+elements of an array, or components of a vector, or columns of a matrix.
+
+This instruction will be specialized to an *OpConstantCompositeContinuedINTEL*
+instruction.
+
+See <<Specialization, Specialization>>.
+
+| Capability:
+*CapabilityLongConstantCompositeINTEL*
+
+| 2 + variable | 6092 | _<id>, <id>, ... Constituents_ |
+|=====
+
+Modify the description of *OpConstantComposite* instruction, adding the
+following sentence to the end:
+In case if it is not possible to specify all the _Constituents_ by one
+*OpConstantComposite* instruction, i.e. if number of members of the
+_Result type_ and corresponding _Constituents_ is greater than maximum possible
+WordCount, the remaining _Constituents_ are specified by following
+*OpConstantCompositeContinuedINTEL* instructions.
+
+Modify the description of *OpSpecConstantComposite* instruction, adding the
+following sentence to the end:
+In case if it is not possible to specify all the _Constituents_ by one
+*OpSpecConstantComposite* instruction, i.e. if number of members of the
+_Result type_ and corresponding _Constituents_ is greater than maximum possible
+WordCount, the remaining _Constituents_ are specified by following
+*OpSpecConstantCompositeContinuedINTEL* instructions.
+
+In Section 3.37.8. Memory Instructions, modify the description of the
+*OpAccessChain* instruction, adding the following sentence to the end:
+In case _Base_ is an *OpConstantComposite* instruction followed by one or
+more *OpConstantCompositeContinuedINTEL* instructions, the bounds must be bounds
+of *OpConstantComposite* and *OpConstantCompositeContinuedINTEL* instructions
+together.
+
+Issues
+------
+
+1) Do we need to define additional validation rules?
+
+Resolution:
+
+No.
+
+2) Do we need modifications of the OpConstantComposite/OpSpecConstantComposite
+instruction description?
+
+Resolution:
+
+Yes, it seems that description of these instructions defines one to one match
+between composite type members and Constituents by the sentence:
+"There must be exactly one Constituent for each top-level
+member/element/component/column of the result." Done.
+
+3) We also might want to modify OpAccessChain to clarify how it works on large
+constants.
+
+Resolution:
+
+Yes. Added statement clarifying bounds of OpAccessChain instruction in case when
+long composite constant is accessed.
+
+Revision History
+----------------
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1|2020-11-09|Mariya Podchishchaeva|Initial revision
+|2|2020-11-13|Mariya Podchishchaeva|Apply comments from AlexeySo and AlexeySa +
```

Comment:
I'd prefer to see my full name as "Alexey Sotkin".

Reply:
Sorry, fixed.